src/submission/submission-service.ts

```diff
@@ -851,26 +859,110 @@ export namespace operations {
     );
   }
 
-  const applyExceptionIfExists = (
-    exceptions: DeepReadonly<ProgramException['exceptions']>,
-    validationError: dictionaryEntities.SchemaValidationError,
-    record: DataRecord,
-  ): boolean => {
-    // missing required field, validate as normal, exceptions still require a submitted value
+  const isException = ({
+    record,
+    exceptions,
+    validationError,
+  }: {
+    record: dictionaryEntities.DataRecord;
+    exceptions: ExceptionRecords;
+    validationError: dictionaryEntities.SchemaValidationError;
+  }): boolean => {
+    // missing required field error, validate as normal, exceptions still require a submitted value
     if (
       validationError.errorType ===
       dictionaryEntities.SchemaValidationErrorTypes.MISSING_REQUIRED_FIELD
     ) {
       return false;
+    } else {
+      // find exception for field
+      const validationErrorFieldName = validationError.fieldName;
+      const exception = findException({ exceptions, record, validationErrorFieldName });
+
+      // check exception value matches error field value
+      return exception?.requested_exception_value === record[validationErrorFieldName];
+    }
+  };
+
+  const findException = ({
+    exceptions,
+    record,
+    validationErrorFieldName,
+  }: {
+    exceptions: ExceptionRecords;
+    record: DataRecord;
+    validationErrorFieldName: string;
+  }): ProgramExceptionRecord | EntityExceptionRecord | undefined => {
+    if (isArrayOfEntityExceptionRecord(exceptions)) {
+      // entity exception
+      return exceptions.find(
+        exception =>
+          exception.requested_core_field === validationErrorFieldName &&
+          findExceptionByEntity({ exception, record }),
+      );
+    } else {
+      // program exception
+      return exceptions.find(exception => {
+        exception.requested_core_field === validationErrorFieldName;
+      });
+    }
+  };
+
+  const specimenKeyField: keyof SpecimenExceptionRecord = 'submitter_specimen_id';
+  const findExceptionByEntity = ({
+    exception,
+    record,
+  }: {
+    exception: EntityExceptionRecord;
+    record: DataRecord;
+  }) => {
+    // NB: keep type checks as string literals for current project TS version 3.9.5
+    if ('submitter_specimen_id' in exception) {
+      return exception[specimenKeyField] === record[specimenKeyField];
+    } else {
+      return false;
```

Comment:
just to be sure I understand why you're not just returning line 921, are you wrapping it in an if/else just as narrowing to stop typescript complaints?

Reply:
the if/else is because there will be further entity exception checks.
eg.
```typscript
 if ('submitter_specimen_id' in exception) {
      return exception[specimenKeyField] === record[specimenKeyField];
    } else if ('submitter_followup_id in exception{
          return exception[followupKeyField] === record[followupKeyField];

    }else{
      return false;
      }
```

Reply:
fair enough! 👍